src/sfctl/custom_app.py

```diff
@@ -68,11 +73,43 @@ def upload_to_fileshare(source, dest, show_progress):
     if show_progress:
         print('Complete', file=sys.stderr)
 
-def upload_to_native_imagestore(sesh, endpoint, abspath, basename, #pylint: disable=too-many-locals
-                                show_progress):
+def get_timeout_left(target_timeout):
+    """
+    Return the number of seconds until timeout is reached, given a target_timeout which represents
+      the time at which the timer should stop. If the time left is less than 0, return 0
+    :param target_timeout: time measured as from epoch in seconds
+    :return: int
+    """
+    current_time = int(time())  # time from epoch in seconds
+    time_left = target_timeout - current_time
+
+    if time_left <= 0:
+        return 0
+    return time_left
+
+def get_lesser(num_a, num_b):
+    """
+    Return the lesser of int num_a and int num_b. If the lesser number is less than 0, return 0
+    :param num_a: (int)
+    :param num_b: (int)
+    :return: Return the smaller of num_a or num_b.
+    """
+
+    if num_a <= 0 or num_b <= 0:
+        return 0
+
+    if num_a <= num_b:
```

Comment:
super small nitpick, this could be return min(num_a,  num_b) and the whole function could be
return max(0 , min(num_a, num_b) ) but that isn't as explicit.